test/e2e/networking.go

```diff
@@ -79,235 +75,172 @@ var _ = framework.KubeDescribe("Networking", func() {
+			for _, endpointPod := range config.endpointPods {
+				config.dialFromTestContainer("http", endpointPod.Status.PodIP, endpointHttpPort, tries, 0, sets.NewString(endpointPod.Name))
 			}
-		}()
-
-		By("Creating a webserver (pending) pod on each node")
-
-		framework.ExpectNoError(framework.WaitForAllNodesSchedulable(f.Client))
-		nodes := framework.GetReadySchedulableNodesOrDie(f.Client)
-		// This test is super expensive in terms of network usage - large services
-		// result in huge "Endpoint" objects and all underlying pods read them
-		// periodically. Moreover, all KubeProxies watch all of them.
-		// Thus we limit the maximum number of pods under a service.
-		//
-		// TODO: Remove this limitation once services, endpoints and data flows
-		// between nodes and master are better optimized.
-		maxNodeCount := 250
-		if len(nodes.Items) > maxNodeCount {
-			nodes.Items = nodes.Items[:maxNodeCount]
-		}
+		})
 
-		if len(nodes.Items) == 1 {
-			// in general, the test requires two nodes. But for local development, often a one node cluster
-			// is created, for simplicity and speed. (see issue #10012). We permit one-node test
-			// only in some cases
-			if !framework.ProviderIs("local") {
-				framework.Failf(fmt.Sprintf("The test requires two Ready nodes on %s, but found just one.", framework.TestContext.Provider))
+		It("should function for intra-pod communication: udp [Conformance]", func() {
+			for _, endpointPod := range config.endpointPods {
+				config.dialFromTestContainer("udp", endpointPod.Status.PodIP, endpointUdpPort, tries, 0, sets.NewString(endpointPod.Name))
 			}
-			framework.Logf("Only one ready node is detected. The test has limited scope in such setting. " +
-				"Rerun it with at least two nodes to get complete coverage.")
-		}
+		})
 
-		podNames := LaunchNetTestPodPerNode(f, nodes, svcname)
+		It("should function for node-pod communication: http [Conformance]", func() {
+			for _, endpointPod := range config.endpointPods {
+				config.dialFromNode("http", endpointPod.Status.PodIP, endpointHttpPort, tries, 0, sets.NewString(endpointPod.Name))
+			}
+		})
 
-		// Clean up the pods
-		defer func() {
-			By("Cleaning up the webserver pods")
-			for _, podName := range podNames {
-				if err = f.Client.Pods(f.Namespace.Name).Delete(podName, nil); err != nil {
-					framework.Logf("Failed to delete pod %s: %v", podName, err)
-				}
+		It("should function for node-pod communication: udp [Conformance]", func() {
+			for _, endpointPod := range config.endpointPods {
+				config.dialFromNode("udp", endpointPod.Status.PodIP, endpointUdpPort, tries, 0, sets.NewString(endpointPod.Name))
 			}
-		}()
+		})
+	})
 
-		By("Waiting for the webserver pods to transition to Running state")
-		for _, podName := range podNames {
-			err = f.WaitForPodRunning(podName)
-			Expect(err).NotTo(HaveOccurred())
-		}
+	// TODO: Remove [Slow] when this has had enough bake time to prove presubmit worthiness.
```

Comment:
This decomposes a SLOW test (1 Describe + 1 It ~20m) into 2 Describes + 15 It blocks each taking < 5m (but still ~20m serially). @kubernetes/sig-testing running this in presubmit should be fine because It runs in parallel right?